internal/ingress/controller/parser/translate.go

```diff
@@ -373,3 +519,39 @@ func knativeSelectSplit(splits []knative.IngressBackendSplit) knative.IngressBac
 	}
 	return res
 }
+
+func pathsFromK8s(path string, pathType networkingv1.PathType) []*string {
+	switch pathType {
+	case networkingv1.PathTypePrefix:
+		base := strings.Trim(path, "/")
+		if base == "" {
+			return kong.StringSlice("/")
+		}
+		return kong.StringSlice(
+			"/"+base+"$",
+			"/"+base+"/",
+		)
+	case networkingv1.PathTypeExact:
+		relative := strings.TrimLeft(path, "/")
+		return kong.StringSlice("/" + relative + "$")
+	default: // includes ImplementationSpecific
+		if path == "" {
+			return kong.StringSlice("/")
+		}
+		return kong.StringSlice(path)
+	}
+}
+
+// HighestPriority is a Kong regex_priority intended to be higher than any other priority generated by KIC.
+const HighestPriority = 1000000
+
+func priorityForPath(path string, pathType networkingv1.PathType) int {
+	switch pathType {
+	case networkingv1.PathTypeExact:
+		return HighestPriority
+	case networkingv1.PathTypePrefix:
+		return len(path)
```

Comment:
Why is this actually required?

Reply:
The Ingress V1 KEP says:

> Path matching semantics
> 
> 4. If there is more than one potential match: 
>     ...
>     ii. For multiple prefix matches, the longest Path p_i will be the matching path.

Putting `len(path)` here gives a longer match priority over a shorter match.

This priority metric would be more natural if we counted path segments instead of characters, but this way the logic is simpler (and equivalent).

That being said, I'm open to a suggestion of any other definition of the priority metric that guarantees the order as specified by the KEP.

Reply:
bd6f0b8

Reply:
Assuming that Kong prefers the longest match if multiple Prefix routes match, we can remove the `len`.

Reply:
Following discussion elsewhere, I think the existing matching behavior in the Kong proxy should handle this: prefix routes already evaluate longer to shorter, with some confusing caveats. https://docs.konghq.com/2.1.x/proxy/#matching-priorities is pretty complex all told.

We should be really cautious about adding magic numbers or similar to enforce K8S-Ingress law over top existing Kong proxy stuff by abusing the latter's tools to inject "this _is_ user intent!" into Kong configuration without somehow calling it out someplace other than code. That's especially true if we're not opening a discussion with the core team.

@guanlan idk really what the best path is to reconcile those sort of differences is (thankfully, here, there actually isn't any as far as I can tell, but this sort of conflict has come up in the past and will continue to arise in the future). I don't think there are great options either way, but we may want to look at starting to maintain a "known areas where K8S Kong defaults differ from stock Kong defaults" reference if nothing else.